src/raftstore/store/peer_storage.rs

```diff
@@ -308,39 +308,78 @@ impl InvokeContext {
     }
 
     #[inline]
-    pub fn save_raft_to(&self, db: &DB, wb: &mut WriteBatch) -> Result<()> {
-        let handle = try!(rocksdb::get_cf_handle(db, CF_RAFT));
-        try!(wb.put_msg_cf(handle,
-                           &keys::raft_state_key(self.region_id),
-                           &self.raft_state));
+    pub fn save_raft_state_to(&self, raft_wb: &mut WriteBatch) -> Result<()> {
+        try!(raft_wb.put_msg(&keys::raft_state_key(self.region_id), &self.raft_state));
         Ok(())
     }
 
     #[inline]
-    pub fn save_apply_to(&self, db: &DB, wb: &mut WriteBatch) -> Result<()> {
-        let handle = try!(rocksdb::get_cf_handle(db, CF_RAFT));
-        try!(wb.put_msg_cf(handle,
-                           &keys::apply_state_key(self.region_id),
-                           &self.apply_state));
+    pub fn save_snapshot_raft_state_to(&self,
+                                       snapshot_index: u64,
+                                       wb: &mut WriteBatch)
+                                       -> Result<()> {
+        let mut snapshot_raft_state = self.raft_state.clone();
+        if snapshot_index < snapshot_raft_state.get_hard_state().get_commit() {
+            snapshot_raft_state.mut_hard_state().set_commit(snapshot_index);
```

Comment:
I think we can set snapshot index here directly, no need to check here. Becuase `snapshot_raft_state.get_hard_state().get_commit()` is always >= snapshot index.